lib/utils.ts

```diff
@@ -1,28 +1,43 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import jsonwebtoken from 'jsonwebtoken'
+import * as t from 'io-ts'
+import { PathReporter } from 'io-ts/lib/PathReporter'
+import { isLeft } from 'fp-ts/lib/Either'
 
 export const isString = (text: any): text is string => {
   return typeof text === 'string' || text instanceof String
 }
 
 export const parseString = (string: any, objectKey: string): string => {
   if (!string || !isString(string)) {
-    throw new Error(`Incorrect or missing ${objectKey}: ${string}`)
+    throw new TypeError(`Incorrect or missing ${objectKey}: ${string}`)
   }
 
   return string
 }
 
-export const parseNumber = (num: any): number => {
+export const parseNumber = (objectKey: string, num: any): number => {
   if (isNaN(num) || !num) {
-    throw new Error(`${num} is not a number`)
+    throw new TypeError(`${objectKey} is not a number: ${num}`)
   }
   return Number(num)
 }
 
 export const parseLocale = (locale: string | string[]): string =>
   locale ? locale.toString() : 'en'
 
+export const validate = <A, I, O>(
+  runtimeType: t.Type<A, O, I>,
+  inputToValidate: I,
+): A => {
+  const errorOrResult = runtimeType.decode(inputToValidate)
+  if (isLeft(errorOrResult)) {
+    throw new TypeError(PathReporter.report(errorOrResult).toString())
```

Comment:
Is this a meaningful error message for the end user?

Reply:
It's not meaningful for end user. We can create new error type for io-ts errors and handle them separately in error handler middleware to send better error message for end user.